neonvm/pkg/ipam/ipam.go

```diff
@@ -233,8 +233,17 @@ func (i *IPAM) acquireORrelease(ctx context.Context, vmName string, vmNamespace
 	case <-leCtx.Done():
 		err = fmt.Errorf("context got timeout while waiting to become leader")
 	}
+	if err != nil {
+		return ip, err
+	}
 
 	wg.Wait()
+
+	// if ip.String() still <nil> then probably something wrong with leader election
+	if len(ip.String()) == 0 {
```

Comment:
> len(...) == 0

typo? should be checking if `ip.String() == "<nil>"` or if `ip == nil` ?

see also: https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/net/ip.go;l=474-487

Reply:
`ip` variable is struct `net.IPNet` and it not comparable directly with `nil`. Also `ip.String()` could be empty string (`""`), I always prefer compare `len() == 0` to check string is nil or empty.

Reply:
ah ok. the reason I say `"<nil>"` specifically is that it's what `ip.String()` returns if there's issues:

```
nn, m := networkNumberAndMask(n)
if nn == nil || m == nil {
    return "<nil>"
}
```

(see the linked source above)

Presumably there's other ways to check if it's valid though?

Reply:
`return "<nil>"` what the... then yes, `len() == 0` is incorrect